camel-component/qa-camel/src/main/java/io/debezium/examples/camel/pipeline/QaDatabaseUserNotifier.java

```diff
@@ -0,0 +1,116 @@
+                header(DebeziumConstants.HEADER_OPERATION).in(
+                        constant(Envelope.Operation.READ.code()),
+                        constant(Envelope.Operation.CREATE.code()));
+        final Predicate isUpdateEvent =
+                header(DebeziumConstants.HEADER_OPERATION).isEqualTo(Envelope.Operation.UPDATE.code());
+        final Predicate isQuestionEvent =
+                header(DebeziumConstants.HEADER_IDENTIFIER).isEqualTo(EVENT_TYPE_QUESTION);
+        final Predicate isAnswerEvent =
+                header(DebeziumConstants.HEADER_IDENTIFIER).isEqualTo(EVENT_TYPE_ANSWER);
+        final Predicate hasManyAnswers =
+                PredicateBuilder.and(
+                        isCreateEvent,
+                        simple("${exchangeProperty[aggregate].answers.size} == 2"));
+
+        final AggregateStore store = new AggregateStore();
+
+        from(ROUTE_MAIL_QUESTION_CREATE)
+            .routeId(QaDatabaseUserNotifier.class.getName() + ".QuestionNotifier")
+            .setHeader("To").simple("${body.email}")
+            .setHeader("Subject").simple("Question created/edited")
+            .setBody().simple("Question '${body.text}' was created or edited")
+            .to(SMTP_SERVER);
+
+        from(ROUTE_MAIL_ANSWER_CHANGE)
+            .routeId(QaDatabaseUserNotifier.class.getName() + ".AnswerNotifier")
+            .setHeader("To").simple("${body.email}; ${exchangeProperty[aggregate].email}")
+            .setHeader("Subject").simple("Answer created/edited")
+            .setBody().simple("Answer '${body.text}' was added/updated for question '${exchangeProperty[aggregate].text}'")
+            .to(SMTP_SERVER);
+
+        from(ROUTE_GET_AGGREGATE)
+            .routeId(QaDatabaseUserNotifier.class.getName() + ".ReadAggregate")
+            .setHeader(InfinispanConstants.KEY).body()
+            .setHeader(InfinispanConstants.OPERATION).constant("GET")
+            .to(ROUTE_STORE_QUESTION_AGGREGATE)
+            .filter(body().isNotNull())
+                .unmarshal().json(JsonLibrary.Jackson, Question.class)
+                .log(LoggingLevel.TRACE, "Unarshalled question ${body}");
+
+        from(ROUTE_WRITE_AGGREGATE)
+            .routeId(QaDatabaseUserNotifier.class.getName() + ".WriteAggregate")
+            .setHeader(InfinispanConstants.KEY).simple("${body.id}")
+            .log(LoggingLevel.TRACE, "About to marshall ${body}")
+            .marshal().json(JsonLibrary.Jackson)
+            .log(LoggingLevel.TRACE, "Marshalled question ${body}")
+            .setHeader(InfinispanConstants.VALUE).body()
+            .to(ROUTE_STORE_QUESTION_AGGREGATE);
+
+        from("debezium-postgres:localhost?"
+                + "databaseHostname=localhost"
+                + "&databaseUser=postgres"
+                + "&databasePassword=postgres"
+                + "&databaseDbname=postgres"
+                + "&databaseServerName=qa"
+                + "&databaseHistoryFileFilename=history"
```

Comment:
Shouldn't be needed for Postgres?

Reply:
Right, it was originally MySQL and I switched to postgres.

Unfortunately for Twitter you need API and for that you need a developer account which you need to request. That's quite a few steps you need take take ahead.

For email sending you need SMTP server and to bypass that you start the mailgo container - so you see which emails and to whom were sent by Camel.

Reply:
The history stuff still should be removed?